### DISCUSSION TITLE
Fetched data keep the order which it was inserted with? (FIFO)

### DISCUSSION THREAD

**Discussion by mbn18** [isar/isar#978] [Help] [answered]
Hey,

Every N seconds the project `putSync`  an item with `timestamp`.

And when there is enough data, the data is fetched out using:
```
i.SomeObject.where().limit(10).findAllSync();
```

And the Question is, will the data get out in `FIFO` style? Meaning the first item that was stored will be the first to be fetched.
Or it is required to `sort` by some field (`timestamp`) to maintain order?

**Comment by simc** [comment, marked as answer]
By default results will be stored in id order. So if you use auto increment ids or your custom ids are in increasing order, Isar will work in FIFO style
